code/controllers/subsystem/spatial_gridmap.dm

```diff
@@ -441,13 +505,65 @@ SUBSYSTEM_DEF(spatial_grid)
 	for(var/list/z_level_grid as anything in grids_by_z_level)
 		for(var/list/cell_row as anything in z_level_grid)
 			for(var/datum/spatial_grid_cell/cell as anything in cell_row)
-				if(to_remove in (cell.hearing_contents | cell.client_contents))
+				if(to_remove in (cell.hearing_contents | cell.client_contents | cell.atmos_contents))
 					containing_cells += cell
 					if(remove_from_cells)
 						force_remove_from_cell(to_remove, cell)
 
 	return containing_cells
 
+/// Adds grid awareness to the passed in atom, of the passed in type
+/// Basically, when this atom moves between grids, it wants to have enter/exit cell called on it
+/datum/controller/subsystem/spatial_grid/proc/add_grid_awareness(atom/movable/add_to, type)
+	// We need to ensure we have a new list reference, to build our new key out of
+	var/list/current_list = spatial_grid_categories[add_to.spatial_grid_key]
+	if(current_list)
+		current_list = current_list.Copy()
+	else
+		current_list = list()
+	// Now we do a binary insert, to ensure it's sorted (don't wanna overcache)
+	BINARY_INSERT_DEFINE(type, current_list, SORT_VAR_NO_TYPE, type, SORT_COMPARE_DIRECTLY, COMPARE_KEY)
+	update_grid_awareness(add_to, current_list)
+
+/// Alerts the atom's current cell that it wishes to be treated as a member
+/// This functionally amounts to "hey, I was recently made aware by [add_grid_awareness], please insert me into my current cell"
+/datum/controller/subsystem/spatial_grid/proc/add_grid_membership(atom/movable/add_to, turf/target_turf, type)
```

Comment:
could you group the add's with remove's instead of add's with add's and remove's with remove's.

Reply:
these procs should also be higher up in the file, id say either just above or just below enter_cell() and exit_cell() as a pair